Source/WebKitLegacy/mac/WebView/WebView.mm

```diff
@@ -2312,10 +2312,8 @@ - (void)_dispatchTileDidDraw:(CALayer*)tile
         return;
     }
 
-    ALLOW_DEPRECATED_DECLARATIONS_BEGIN
-    if (!OSAtomicCompareAndSwap32(0, 1, &_private->didDrawTiles))
+    if (!WTF::atomicCompareExchangeWeakRelaxed(&_private->didDrawTiles, NO, YES))
```

Comment:
These two functions are not equivalent.

Reply:
Examining the definitions of both functions, yes they are equivalent.

Reply:
Can you post the definitions you examined, and define the architectures on which you think they're equivalent?

Reply:
Yes. Here 
<img width="923" alt="Screen Shot 2023-02-21 at 8 53 29 AM" src="https://user-images.githubusercontent.com/83477269/220364621-4c47d04f-6958-40f7-8176-13e5c97e9bf9.png">
<img width="923" alt="Screen Shot 2023-02-21 at 8 53 12 AM" src="https://user-images.githubusercontent.com/83477269/220364625-8de8fc00-f36b-4e99-94cf-1909000aacc2.png">
<img width="1096" alt="Screen Shot 2023-02-21 at 8 51 26 AM" src="https://user-images.githubusercontent.com/83477269/220364628-af53b42d-2cdb-459c-91f3-1c4a189151ba.png">

Reply:
When you say "I think strong is overkill", does that mean that you now see that these functions are not equivalent?

If so, can you explain the difference between these functions, and why you believe that difference is OK?

Reply:
Yes, one has a strong atomic compare which affects memory sequence, and the one I chose is more relaxed since the original code did not write OSAtomicCompare with the barrier.

Reply:
No, that's not right. Please re-read the documentation on compare_exchange_weak.